### DISCUSSION TITLE
Debian BullsEye Docker Installation

### DISCUSSION THREAD

**Discussion by cmatte** [awawa-dev/HyperHDR#594] [General]
Hi there,

I've been trying to install HyperHDR within the docker debian:bullseye image with partial success on a Synology NAS DS923+ (x86_64), learning from the docker-compose from HyperionNG [here](https://github.com/psychowood/hyperion.ng-docker/blob/main/docker-compose.standalone.yml) as well as directions [here](https://awawa-dev.github.io/). The issue I am looking at is that the webpage is just showing language tags e.g. dashboard_message_default_password_t - as well as the below error in the log. Full log also attached in .txt format (it is .html - yet GitHub does not support the extension so please rename for readability or open with a notepad). 

1. Can anybody advise how to amend the docker compose below to address the little (apparently) trouble?
2. Is there a supported way not to run HyperHDR as root? 

Log error:
`/var/lib/dpkg/info/hyperhdr.postinst: 32: root: not found`

Full log:
[hyperhdr.txt](https://github.com/awawa-dev/HyperHDR/files/11995421/hyperhdr.txt)

Docker-compose.yaml:

```
version: '3.3'

services:
  hyperhdr:
    image: debian:bullseye
    container_name: hyperhdr
    command: bash -c "groupadd -f hyperhdr || true &&
                    adduser -q --uid ${UID:-1000} --gid ${GID:-1000} --disabled-password --no-create-home hyperhdr || true &&
                    mkdir -p /config &&
                    chown ${UID:-1000}:${GID:-1000} /config &&
                    apt-get update &&
                    apt-get upgrade &&
                    apt-get install wget gpg sudo curl dpkg lsb-release -y &&
                    wget -qO /usr/share/keyrings/hyperhdr.public.apt.gpg.key https://awawa-dev.github.io/hyperhdr.public.apt.gpg.key &&
                    sudo chmod go+r /usr/share/keyrings/hyperhdr.public.apt.gpg.key &&
                    echo \"deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hyperhdr.public.apt.gpg.key] https://awawa-dev.github.io $(lsb_release -cs) main\" > /etc/apt/sources.list.d/hyperhdr.list &&
                    sudo apt-get update &&
                    sudo apt-get install hyperhdr -y &&
                    sudo apt-get upgrade &&
                    sudo apt-get clean &&
                    sudo -u hyperhdr /usr/bin/hyperhdr -v --service -u /config"
    ports:
      - "19400:19400"
      - "19444:19444"
      - "19445:19445"
      - "8090:8090"
      - "8092:8092"
    volumes:
      - /volume1/docker/hyperhdr/config:/config
      - /volume1/docker/hyperhdr/webconfig:/webconfig
    restart: unless-stopped
```

**Comment by awawa-dev** [comment]
Try this. It works for me. 
```
version: '3.3'

services:
  hyperhdr:
    image: debian:bullseye
    container_name: hyperhdr
    command: bash -c "groupadd -f hyperhdr || true &&
                    adduser -q --uid ${UID:-1000} --gid ${GID:-1000} --no-create-home --disabled-password hyperhdr || true &&
                    chown ${UID:-1000}:${GID:-1000} /home/hyperhdr &&
                    apt-get update &&
                    apt-get upgrade &&
                    apt-get install wget gpg sudo curl dpkg lsb-release -y &&
                    wget -qO /usr/share/keyrings/hyperhdr.public.apt.gpg.key https://awawa-dev.github.io/hyperhdr.public.apt.gpg.key &&
                    sudo chmod go+r /usr/share/keyrings/hyperhdr.public.apt.gpg.key &&
                    echo \"deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hyperhdr.public.apt.gpg.key] https://awawa-dev.github.io $(lsb_release -cs) main\" > /etc/apt/sources.list.d/hyperhdr.list &&
                    apt-get update &&
                    apt-get install hyperhdr -y &&
                    apt-get clean &&
                    sudo -u hyperhdr /usr/bin/hyperhdr -v --service"
    ports:
      - "19400:19400"
      - "19444:19444"
      - "19445:19445"
      - "8090:8090"
      - "8092:8092"
    volumes:
      - /volume1/docker/hyperhdr/config:/home/hyperhdr
    restart: unless-stopped
```
BTW `sudo -u hyperhdr` so it runs as non-root

**Reply by cmatte** [reply]
Thanks! Works well, indeed :)

**Reply by fightforlife** [reply]
I am also using this, thank you very much!
But for me the Log window is always empty, is this expected?

**Comment by awawa-dev** [comment]
@fightforlife 
The "verbose" (-v) option did not work properly in 20beta1, this will be fixed in 20beta2. However, HyperHDR should work despite this with no logs on output.

**Comment by brinza666** [comment]
Maybe anyone could help please? I'm using same .yaml file, it worked perfectly till couple of days. 
Log:
2024/07/14 21:04:35 | stdout | Do you want to continue? [Y/n] Abort.
2024/07/14 21:04:35 | stdout | After this operation, 0 B of additional disk space will be used.
2024/07/14 21:04:35 | stdout | Need to get 709 kB of archives.
2024/07/14 21:04:35 | stdout | 4 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
2024/07/14 21:04:35 | stdout | libgssapi-krb5-2 libk5crypto3 libkrb5-3 libkrb5support0
2024/07/14 21:04:35 | stdout | The following packages will be upgraded:
2024/07/14 21:04:35 | stdout | Calculating upgrade...
2024/07/14 21:04:35 | stdout | Reading state information...
2024/07/14 21:04:35 | stdout | Building dependency tree...
2024/07/14 21:04:35 | stdout | Reading package lists...
2024/07/14 21:04:34 | stdout | Reading package lists...
2024/07/14 21:04:33 | stdout | Hit:3 http://deb.debian.org/debian bullseye-updates InRelease
2024/07/14 21:04:33 | stdout | Hit:2 http://deb.debian.org/debian-security bullseye-security InRelease
2024/07/14 21:04:32 | stdout | Hit:1 http://deb.debian.org/debian bullseye InRelease
2024/07/14 21:04:32 | stderr | adduser: The user `hyperhdr' already exists.

Same sequence repeating... and container crashing/stoping

**Reply by brinza666** [reply]
i'll answer to myself, this solved my issue : 

The primary issue you are facing with the provided Docker Compose file is that the command for the container runs interactively and requires user input for confirmations during package installation. To resolve this, you need to automate these confirmations and handle the case where the user hyperhdr already exists more gracefully.

Here is the corrected Docker Compose YAML script:

yaml

```
version: '3.3'

services:
  hyperhdr:
    image: debian:bullseye
    container_name: hyperhdr
    command: bash -c "
      groupadd -f hyperhdr &&
      (id -u hyperhdr &>/dev/null || adduser --uid ${UID:-1000} --gid ${GID:-1000} --no-create-home --disabled-password hyperhdr) &&
      chown ${UID:-1000}:${GID:-1000} /home/hyperhdr &&
      apt-get update &&
      DEBIAN_FRONTEND=noninteractive apt-get upgrade -y &&
      DEBIAN_FRONTEND=noninteractive apt-get install -y wget gpg sudo curl dpkg lsb-release &&
      wget -qO /usr/share/keyrings/hyperhdr.public.apt.gpg.key https://awawa-dev.github.io/hyperhdr.public.apt.gpg.key &&
      chmod go+r /usr/share/keyrings/hyperhdr.public.apt.gpg.key &&
      echo \"deb [arch=\$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hyperhdr.public.apt.gpg.key] https://awawa-dev.github.io \$(lsb_release -cs) main\" > /etc/apt/sources.list.d/hyperhdr.list &&
      apt-get update &&
      DEBIAN_FRONTEND=noninteractive apt-get install -y hyperhdr &&
      apt-get clean &&
      exec sudo -u hyperhdr /usr/bin/hyperhdr -v --service
    "
    ports:
      - "19400:19400"
      - "19444:19444"
      - "19445:19445"
      - "8090:8090"
      - "8092:8092"
    volumes:
      - /volume1/docker/hyperhdr/config:/home/hyperhdr
    restart: unless-stopped
```
Changes and improvements made:

Automated Package Installation: Added DEBIAN_FRONTEND=noninteractive before apt-get commands to avoid interactive prompts.
User Existence Check: Replaced adduser command with a check for user existence using id -u hyperhdr to avoid errors if the user already exists.
Use exec: Changed the final command to use exec to replace the shell with the hyperhdr process. This ensures proper signal handling within the container.
This should resolve the issues you encountered and provide a more robust solution for running the HyperHDR service in a Docker container.
